version: '3.7'
services:
  api:
    image: ddd_edd_practice/api:latest
    build:
      context: .
      dockerfile: deploy/api/docker/Dockerfile
    volumes:
        - ./:/srv/
    ports:
      - "5000:${DDD_EDD_PRACTICE_API_PORT}"
    container_name: "ddd_edd_practice-api"
    restart: on-failure
    depends_on:
      - db
    env_file:
      - .env
    networks:
      - ddd_edd_practice-network

  db:
    image: postgres:11.6
    user: ${USER_ID}:${USER_ID}
    restart: always
    container_name: "ddd_edd_practice-db"
    environment:
      - POSTGRES_USER=${DDD_EDD_PRACTICE_DB_USER}
      - POSTGRES_PASSWORD=${DDD_EDD_PRACTICE_DB_PASSWORD}
      - POSTGRES_DB=${DDD_EDD_PRACTICE_DB_NAME}
    ports:
      - "54320:${DDD_EDD_PRACTICE_DB_PORT}"
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
      - ./deploy/local_db:/docker-entrypoint-initdb.d
    networks:
      - ddd_edd_practice-network

networks:
  ddd_edd_practice-network:
    driver: bridge
